### PR TITLE
Adding initial draft for Contributing Components and Features to ODH

### PR DESCRIPTION
**Pull request by nakfour**
This PR adds a new document that describes the guidelines to adding new components and large features to ODH.

### CODE REVIEW

**Comment by nakfour** [comment]
from @shgriffi 
1. Must provide example use cases of how the proposed PR helps progress the overall end to end AI/ML objective of ODH.  These can be vertical or horizontal use cases, but needs to show how whatever the PR is fits and integrates with the rest of the ODH stack to tell a complete story.
2. Must have automated CI tests

**Comment by LaVLaS** [comment]
Prior to the proposal doc, what are the software requirements for actual component/feature
* 100% Open Source with well defined LICENSE
* All container images must have readily available build files publicly available
...

**Comment by crobby** [comment]
I think we should also include some requirements around testing.  Any new components must include some degree of testing that is integrated with our testing framework.

**Comment by nakfour** [comment]
@crobby added a note on testing

File: GuidelinesForNewComponents.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Guidelines for Contributing New Components to Open Data Hub
+
+This document describes the guidelines to submit new ideas such as new features and components to Open Data Hub. This specifically pertains to new components or large features that require an architecture review or a re-design of existing components. This documents does not apply to small or incremental features, for those please open an issue with a "feature" label in the [Open Data Hub project](https://issues.redhat.com/projects/ODH/issues/ODH-380?filter=allopenissues)
```
Comment (LaVLaS) [inline GuidelinesForNewComponents.md:3]: You can remove `ODH-380` from the url. It looks like Jira will automatically redirect to the first open issue with this filter enabled

File: GuidelinesForNewComponents.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+## Basic Requirements for Component
+Proposed components must have the following basic requirements:
+1. The component is an open source component with a valid open source license.
+2. The component should be aligned with Open Data Hub goals, architecture and roadmap.
+3. All build and source files for the component must be publicly available. 
+4. For component images and builds it is recommended to use [Red Hat Universal Base Images](https://catalog.redhat.com/software/containers/search?q=ubi&p=1)(UBI) based images.
+5. The component must run on Openshift or OKD platform. 
+
+## Create a Proposal document
+The first step in submitting a new component or feature in Open Data Hub is to create a proposal document and place that in the proposal folder. Please name the file accordingly: <component-name>.md.
```
Comment (LaVLaS) [inline GuidelinesForNewComponents.md:14]: Will the `proposal` folder be located in this repo? If so, should we add the folder and a template to this PR?

File: GuidelinesForNewComponents.md (hunk 1 above)
Comment (vpavlin) [reply GuidelinesForNewComponents.md:3]: Also there is a typo: `This documents does not` -> `This document does not`

File: GuidelinesForNewComponents.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# Guidelines for Contributing New Components to Open Data Hub
+
+This document describes the guidelines to submit new ideas such as new features and components to Open Data Hub. This specifically pertains to new components or large features that require an architecture review or a re-design of existing components. This documents does not apply to small or incremental features, for those please open an issue with a "feature" label in the [Open Data Hub project](https://issues.redhat.com/projects/ODH/issues/ODH-380?filter=allopenissues)
+
+## Basic Requirements for Component
```
Comment (vpavlin) [inline GuidelinesForNewComponents.md:5]: for **a** Component

File: GuidelinesForNewComponents.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+Proposed components must have the following basic requirements:
+1. The component is an open source component with a valid open source license.
```
Comment (vpavlin) [inline GuidelinesForNewComponents.md:7]: Component is there twice -> `is an open source project`

File: GuidelinesForNewComponents.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+2. The component should be aligned with Open Data Hub goals, architecture and roadmap.
```
Comment (vpavlin) [inline GuidelinesForNewComponents.md:8]: Should there be links to architecture and roadmap?

File: GuidelinesForNewComponents.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+3. All build and source files for the component must be publicly available. 
+4. For component images and builds it is recommended to use [Red Hat Universal Base Images](https://catalog.redhat.com/software/containers/search?q=ubi&p=1)(UBI) based images.
```
Comment (vpavlin) [inline GuidelinesForNewComponents.md:10]: Add a space between the url and `(UBI)` - it is rendered like this at the moment:  
> Red Hat Universal Base Images(UBI)

File: GuidelinesForNewComponents.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+5. The component must run on Openshift or OKD platform.
```
Comment (vpavlin) [inline GuidelinesForNewComponents.md:11]: OpenShift (capital S)

File: GuidelinesForNewComponents.md (hunk 2 above)
Comment (vpavlin) [reply GuidelinesForNewComponents.md:14]: +1 to @LaVLaS's suggestion

typo: or **a** feature

File: GuidelinesForNewComponents.md [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+3. All build and source files for the component must be publicly available. 
+4. For component images and builds it is recommended to use [Red Hat Universal Base Images](https://catalog.redhat.com/software/containers/search?q=ubi&p=1)(UBI) based images.
+5. The component must run on Openshift or OKD platform. 
+
+## Create a Proposal document
+The first step in submitting a new component or feature in Open Data Hub is to create a proposal document and place that in the proposal folder. Please name the file accordingly: <component-name>.md. 
+The following information must be included in the proposal document
+1. High level description of the component along with a high level architecture diagram.
+2. A detailed description of the user use cases this components will provide. 
+3. Lower level implementation design and details that include information such as integration implementation details, artifacts, endpoints security considerations, multi-user support, authentication, etc.
```
Comment (vpavlin) [inline GuidelinesForNewComponents.md:18]: Is it "endpoints security considerations" or is there a comma missing - "endpoints, security considerations"?

File: GuidelinesForNewComponents.md [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+
+## Present your Proposal to ODH Community Meeting
+Present your proposal to the Open Data Hub Community in one of the community meetings. You can add your item to the agenda by editing the [wiki](https://github.com/opendatahub-io/opendatahub-community/wiki/Open-Data-Hub-Community-Meeting-Agenda) page and attending the appropriate ODH community meeting in this [calendar](https://github.com/opendatahub-io/opendatahub-community). 
+
+## Open an Issue in Open Data Hub project
+Open an Issue in the Open Data Hub [Jira project](https://issues.redhat.com/secure/RapidBoard.jspa?rapidView=8528&projectKey=ODH&view=planning.nodetail&issueLimit=100) describing the proposed component and pointing to the proposal documentation. In this issue the Open Data Hub community will discuss the new component and decide whether or not to move forward. 
+
+## Submit a Pull Request After Approval
```
Comment (vpavlin) [inline GuidelinesForNewComponents.md:26]: typo: 2 dots after framework

File: GuidelinesForNewComponents.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+Once the opened issue is resolved and there is approval from the Open Data Hub community, a pull request with the implementation details can be submitted. A pull request must include comprehensive unit tests for the component integrated with Open Data Hub testing framework.. If the component includes a UI element, integration with the Open Data Hub dashboard must also be included. It is also required that all new components expose a Prometheus endpoint for metric gathering.
```
Comment (vpavlin) [inline GuidelinesForNewComponents.md:27]: typo: 2 dots after framework

File: README.md [hunk 11]
```diff
@@ -1,18 +1,22 @@
-# Open Data Hub Community Meetings
+# Open Data Hub Community
+
+## Open Data Hub Guidelines
+To contribute new ideas such as new features or components to Open Data Hub please follow the new component proposal guidelines.
```
Comment (vpavlin) [inline README.md:4]: Can you add link to the guidelines?

File: GuidelinesForNewComponents.md (hunk 5 above)
Comment (nakfour) [reply GuidelinesForNewComponents.md:8]: We dont have those yet, we can add the link once we create the docs

File: GuidelinesForNewComponents.md (hunk 2 above)
Comment (nakfour) [reply GuidelinesForNewComponents.md]: added a folder

**Review by LaVLaS — APPROVED** [review]
/lgtm

**Pull request merged**